packages/dds/tree/src/test/utils.ts

```diff
@@ -1300,3 +1302,23 @@ export function moveWithin(
 ) {
 	editor.move(field, sourceIndex, count, field, destIndex);
 }
+
+/**
+ * Clones a batch of revertibles for a target branch.
+ * @param revertibles - Array of revertibles to clone
+ * @param targetBranch - The target branch to clone the revertibles for
+ * @returns Array of cloned revertibles, maintaining the same order as the input
+ */
+export function cloneRevertiblesInBatch(
+	revertibles: RevertibleAlpha[],
+	targetBranch: TreeBranch,
+): RevertibleAlpha[] {
+	const disposedRevertible = revertibles.find(
+		(revertible) => revertible.status === RevertibleStatus.Disposed,
+	);
+	if (disposedRevertible !== undefined) {
```

Comment:
How does this batch clone operation recover if this list has 1 revertible that's disposed? Are you supposed to abandon the clone operation for every other revertible in the list?

Reply:
Good question. 

Yes, for now that is indeed the intention. If there is _any disposed_ `Revertible` object, the entire batched clone should fail. 

This is because `Revertible` that has been disposed isn't re-usable, and user should pay attention to the life management. AFAIK, I think the entire clone operation should fail and throw an error instead of returning the list of cloned revertibles that are clonable. 

And this is also one of the requirements mentioned in the task:
_"passing in an invalid revertible errors without forking any other passed in revertibles, or at least making sure any already forked revertibles are disposed"_